pkg/resource/replication_group/delta_util.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package replication_group
+
+import (
+	"strings"
+	"regexp"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+
+	"github.com/aws-controllers-k8s/elasticache-controller/pkg/common"
+)
+
+// remove non-meaningful differences from delta
+func filterDelta(
+	delta *ackcompare.Delta,
+	desired *resource,
+	latest *resource,
+) {
```

Comment:
Nice.

Not related to this PR but mentioning it here due to its context: I think this is where Server defaults can be handled: Remove differences from delta when corresponding spec field is nil if last applied spec also had that field nil.
* This will require access to "last applied spec" which needs to be kept somewhere for reference.

Reply:
Yep this should enable us to (at least temporarily) handle defaults before the framework-level solution is ready